src/main/resources/META-INF/resources/primefaces/overlaypanel/overlaypanel.js

```diff
@@ -428,20 +410,10 @@ PrimeFaces.widget.OverlayPanel = PrimeFaces.widget.DynamicOverlayWidget.extend({
      */
     setupDialogSupport: function() {
         if (this.target && this.target[0]) {
-            var dialog = this.target[0].closest('.ui-dialog');
-            if (dialog) {
-                var $dialog = $(dialog);
-                if ($dialog.length == 1) {
-                    //set position as fixed to scroll with dialog
-                    if($dialog.css('position') === 'fixed') {
-                        this.jq.css('position', 'fixed');
-                    }
-
-                    //append to body if not already appended by user choice
-                    if(!this.cfg.appendTo) {
-                        this.jq.appendTo(document.body);
-                    }
-                }
+            this.cfg.appendTo = PrimeFaces.utils.resolveAppendTo(this, this.target, this.jq);
+            //append to body if not already appended by user choice
+            if(!this.cfg.appendTo) {
```

Comment:
i think i understood this now.
if we manually appendTo body now, without setting this.cfg.appendTo=@body, our DynamicOverlayWidget logic will never remove it.
Can you check why your integration tests doesnt fail? Normally if you set appendTo=null and update it, there should be 2 overlays below the body?

Reply:
So here is what I can tell is happening and maybe i am confused so forgive me.

if appendTo== null then we automatically append to the body.
DyanmicOverlayWidget always registers `refresh` and `destroy` methods.
Your `refresh` method calls `cleanupDynamicOverlay` which will remove the overlay only if `appendTo` is actually set else it leaves it in the DOM.
If you change the appendTo with an update the refresh method should destroy the old one still only leaving one in the DOM right?

Reply:
So previously my `destroy` test was saying the overlay was destroyed when appendTo was null and now its saying its not its still in the body.  So I updated my test thinking this behavior is wrong.

Reply:
So here is what happens when appendTo==null
```java
@Test
    @Order(3)
    @DisplayName("OverlayPanel: Destroy overlay")
    public void testDestroy(Page page) {
        // Arrange
        OverlayPanel overlayPanel = page.overlayPanel;
        overlayPanel.show();
        Assertions.assertTrue(overlayPanel.isDisplayed());

        // Act
        page.btnDestroy.click();

        // Assert
        Assertions.assertFalse(overlayPanel.isDisplayed());
        assertConfiguration(overlayPanel.getWidgetConfiguration());
    }
```

Hree is what happens when `appendTo="@(body)"` or if used inside a dialog which sets the `@(body)`
```java
 @Test
    @Order(4)
    @DisplayName("OverlayPanel: Dialog Destroy overlay")
    public void testDialogDestroy(Page page) {
        // Arrange
        OverlayPanel overlayPanel = page.overlayPanel;
        page.btnDialog.click();
        Assertions.assertTrue(page.dialog.isDisplayed());
        overlayPanel.show();
        Assertions.assertTrue(overlayPanel.isDisplayed());

        // Act
        page.btnDestroy.click();

        // Assert
        try {
            overlayPanel.isDisplayed();
            Assertions.fail("OverlayPanel should have been destroyed.");
        }
        catch (NoSuchElementException ex) {
            // overlay panel should be destroyed
        }
    }
```

Reply:
btnDestroy destroys the overlaypanel with updating a container and set rendered false to overlaypanel?
Then please check in the first test (appendTo==null), that there is now appended overlay left below the body

Reply:
yes appendTo==null the Overlay remains and its not destroyed.

appendTo==@Body the overlay is destroyed

Reply:
Oh actually let me check if it sets Rendered=false

Reply:
OK yes the test calls Destroy and updated the entire `@form` and the overlay `rendered="false"` and its still in the DOM.

Reply:
yep, we should default then to "@(body)" via stateHelper if that should be the default?

Reply:
I think that is what we have been doing everywhere else let me try it.

Reply:
OK with latest changes now in the OverlayPanelBase is defaulted to `"@(body)"` like  we do in other components. this also lets me remove the part in setupDialogSupport about appending it to the body if NULL since `DynamicOverlayWidget` is already doing that with `registerDynamicOverlay`.

However just checking in removeDynamicOverlay...
```javascript
removeDynamicOverlay: function(widget, overlay, overlayId, appendTo) {
            appendTo.children("[id='" +  overlayId + "']").not(overlay).remove();
        }
```

This says only remove the over lay if the appenTo has this id but its NOT already the overlay we have.  Is that correct?

Reply:
i think null is passed on destruction

Reply:
I have never used lightbox but it does seem like a prime candidate to extend DynamicOverlay.  Let me try it with the showcase.